openmls/src/key_packages/mod.rs

```diff
@@ -209,10 +215,42 @@ impl KeyPackage {
         self.payload.extensions.as_slice()
     }
 
+    /// Check whether the this key package supports all the required extensions
+    /// in the provided list.
+    pub fn check_extension_support(
+        &self,
+        required_extensions: &[ExtensionType],
+    ) -> Result<(), KeyPackageError> {
+        let my_extension_types = self.extensions().iter().map(|ext| ext.extension_type());
+        for required in required_extensions.iter() {
+            if !my_extension_types.clone().any(|e| &e == required) {
```

Comment:
Instead of cloning here, it might be cheaper to clone the iterator beforehand.

Reply:
This is cloning the iterator. `any` mutates the iterate that's why we need a fresh one each time (unless we sort first).